src/workflows/trades.ts

```diff
@@ -1,16 +1,16 @@
-import { Signer } from '@ethersproject/abstract-signer';
-import { StarkWallet } from '../types';
+import { L1Signer, L2Signer } from '../types';
 import { GetSignableTradeRequest, TradesApi } from '../api';
-import { serializeSignature, sign, signRaw } from '../utils';
+import { signRaw } from '../utils';
 
 export async function createTradeWorkflow(
-  signer: Signer,
-  starkWallet: StarkWallet,
+  l1Signer: L1Signer,
```

Comment:
This is a breaking change, and with these changes, the parameters put in **the Trading workflow** would differ from the other Workflows. so we need an alignment here to think about our approach for those changes and the approach for release

We would like to finish the workflow params change in 2 tickets:
https://immutable.atlassian.net/browse/WT-423(This PR scope )
https://immutable.atlassian.net/browse/WT-374

So my question is :
* Whether we want to keep the backward capability for some time?
* Whether we want to keep the backward capability for at least workflow params changes?
* If we don't need the backward capability, what's the release plan?

@bruno-imx @zatlite @RenatoChitolina @vmiguez-immutable @lrepolho

Reply:
Here is my thoughts:

> Whether we want to keep the backward capability for some time?
If we can keep backward compatibility, we should keep it for some time and also write about when this support ends. 

> Whether we want to keep the backward capability for at least workflow params changes?
Yes, I think we want to keep backward compatibility for all changes where possible.

> If we don't need the backward capability, what's the release plan?
If backward compatibility is not possible, then we should publish a major version as Core SDK follows semantic versioning. The changelog should include notes about breaking changes and how to migrate.